internal/utils/validation/cd.go

```diff
@@ -94,10 +93,22 @@ func ClusterDeployCredential(ctx context.Context, cl client.Client, cd *kcmv1.Cl
 		return nil, fmt.Errorf("the Credential %s is not Ready", credKey)
 	}
 
-	return cred, isCredIdentitySupportsClusterTemplate(cred, clusterTemplate)
+	return cred, isCredIdentitySupportsClusterTemplate(ctx, cl, cred, clusterTemplate)
 }
 
-func isCredIdentitySupportsClusterTemplate(cred *kcmv1.Credential, clusterTemplate *kcmv1.ClusterTemplate) error {
+func getProviderClusterIdentityKinds(ctx context.Context, cl client.Client, infrastructureProviderName string) []string {
+	obj := &kcmv1.PluggableProvider{}
+
+	if err := cl.Get(ctx, client.ObjectKey{
+		Name: strings.TrimPrefix(infrastructureProviderName, kcmv1.InfrastructureProviderPrefix),
```

Comment:
perf: nothing to trim in this context: the parameter already implies there is no prefix

Reply:
initially, there was no need to trim given that the previous code block had the line `infraProviderName := providerName[len(providers.InfraPrefix):]`, which has been removed for some reason. due to the forced merge that removal led to #1438

Reply:
anyway, it was addressed